nova/cmd/manage.py

```diff
@@ -829,10 +830,15 @@ def purge_request_specs(self, dry_run=False, older_than=90, max_number=0,
         max_number = int(max_number)
 
         admin_context = context.get_admin_context()
-        # There's no need to join the default security_groups and
-        # instance_info_caches tables, hence explicit empty 'columns_to_join'.
-        extant_instance_uuids = [i.get('uuid') for i in db.instance_get_all(
-            admin_context, columns_to_join=[]) if i.get('uuid') is not None]
+        for cell in cell_mapping_obj.CellMappingList.get_all(admin_context):
+            with context.target_cell(admin_context, cell) as cell_context:
+                # There's no need to join the default security_groups and
+                # instance_info_caches tables, hence explicit empty
+                # 'columns_to_join'.
+                extant_instance_uuids |= \
+                    set(i.get('uuid') for i in db.instance_get_all(
+                        cell_context, columns_to_join=[])
```

Comment:
Should raise `NameError` for undefined `extant_instance_uuids`.

Reply:
... That's what I get for moving fast. :/